extensions/ql-vscode/src/view/result-tables.tsx

```diff
@@ -244,6 +269,40 @@ export class ResultTables
       }
     </div>;
   }
+
+  handleMessage(msg: IntoResultsViewMsg): void {
+    switch (msg.t) {
+      case 'untoggleShowProblems':
+        this.setState({
+          problemsViewSelected: false
+        });
+        break;
+
+      default:
+        // noop
+    }
+  }
+
+  // TODO: Duplicated from results.tsx consider a way to
```

Comment:
Can we access the parent's `vscodeMessageHandler`?

Reply:
Not directly. We would need to pass it through the properties. And we'd still need to make calls on (un-)mount. So, any solution I could come up with where we don't duplicate code includes more boilerplate than we have now.

Reply:
We could just extract the message handler into a react context that is shared throughout the react tree.

It would still be more code/boilerplate than we have now. But, at least it would be using proper react idioms. Though, I'd still argue it's more complex than what we have now. Can we consider this a future enhancement in case we need to share the message handler code in more locations?

Reply:
Absolutely. Just curious how this might work, no reason to block this PR.